js/hatrac.js

```diff
@@ -430,13 +432,26 @@ var ERMrest = (function(module) {
             var headers = module.getResponseHeader(response);
             var md5 = headers["content-md5"];
             var length = headers["content-length"];
+            var contentDisposition = headers["content-disposition"];
 
             // If the file is not same, then simply resolve the promise without setting completed and jobDone
             if ((md5 != self.hash.md5_base64) || (length != self.file.size)) {
                 deferred.resolve(self.url);
                 return;
             }
 
+            // hatrac only supports `filename*=UTF-8''` in the content disposition so check for this string to get the filename
+            var contentDispositionPrefix = "filename*=UTF-8''";
+            var filenameIndex = contentDisposition.indexOf(contentDispositionPrefix) + contentDispositionPrefix.length;
+
+            // check if filename in content disposition is different from filename being uploaded
+            // if it is, instead set a different flag and don't mark it as done 
```

Comment:
Can't we send the put request to update the metadata here? That way chaise doesn't need to know anything about this filename/metadata change. From chaise perspective, the file exists and so it can just skip creating the upload job.